nexus/types/src/external_api/shared.rs

```diff
@@ -151,6 +151,16 @@ pub enum UserProvisionType {
     Jit,
 }
 
+/// The service intended to use this certificate.
+#[derive(Clone, Copy, Debug, Deserialize, Serialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ServiceUsingCertificate {
+    /// This certificate is intended for the "Nexus" service.
+    ///
+    /// This service acts as the main API point for accessing the Oxide rack.
+    Nexus,
```

Comment:
nit: I'd suggest we call this `External` or `ExternalApi` instead.  I'm not sure operators should be aware of "Nexus", or that it's the external endpoint (it might not always be! if we add a load balancer in the future).  Plus, these certs are used for only one of Nexus's servers (not the internal API).